src/ray/core_worker/transport/direct_actor_transport.cc

```diff
@@ -117,6 +117,7 @@ void CoreWorkerDirectActorTaskSubmitter::ConnectAndSendPendingTasks(
     auto status =
         PushTask(*client, request, TaskID::FromBinary(request.task_spec().task_id()),
                  request.task_spec().num_returns());
+    RAY_CHECK_OK(status);
```

Comment:
Does this mean that the first time a driver tries to connect to a direct call actor, if the actor is already dead, then the whole driver will fail? That doesn't seem like the right behavior.

Reply:
No. `PushTask` now always return ok. If the actor is dead, we'll treat the task as failed, and use an app-level exception to inform callers.